libraries/AC_ADRC/AC_ADRC.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Param/AP_Param.h>
+
+
+class AC_ADRC {
+public:
+    AC_ADRC(float B0, float dt);
+
+    CLASS_NO_COPY(AC_ADRC);
+
+    float update_all(float target, float measurement, bool limit);
+    void reset_eso(float measurement);
+    float fal(float e, float alpha, float delta);
+    float sign(float x);
+    // Reset filter
+    void reset_filter(){
+        _flags.reset_filter = true;
+    }
+
+    // flags
+    struct ap_adrc_flags {
+        bool reset_filter :1; // true when input filter should be reset during next call to set_input
```

Comment:
I copied ADRC library implementation from https://github.com/ArduPilot/ardupilot/pull/20243. Maybe it will be better to skip the example backends(only leave PID perhaps?) implementation until this PR goes in.